### ISSUE TITLE
Nemotron-Nano-9b-V2 has low full-weights SFT accuracy

### ISSUE THREAD

**Issue by sabhatinas** [NVIDIA-NeMo/Automodel#893]
**Describe the bug**

Nemotron-Nano-9b-V2 full weights SFT accuracy is really low. The rouge score is 0.07. However, the older AutoModel training scripts give a higher rouge score (0.23) for the same hyper-parameters.

**Steps/Code to reproduce bug**

torchrun --nproc-per-node=8 /opt/Automodel/examples/llm_finetune/finetune.py --config config.yaml

Config.yaml - [config.yaml.rtf](https://github.com/user-attachments/files/23777498/config.yaml.rtf)

Dataset is attached.

[email-composition-train.tar.gz](https://github.com/user-attachments/files/23777450/email-composition-train.tar.gz)

**Expected behavior**

Base model rouge score is at least 0.23. Full weights SFT should give at least 0.23.

**Additional context**
